app/src/org/odk/collect/android/views/media/MediaLayout.java

```diff
@@ -262,7 +244,7 @@ public void onClick(View v) {
                         errorMsg = getContext().getString(R.string.file_invalid, imageFile);
 
                     }
-                } else if (errorMsg == null) {
```

Comment:
this null check is always true, looks like we just want an unconditional else statement here.